### PR TITLE
Added "Clear Files" button to Area Upload Manager to reset selected files

### PR DESCRIPTION
**Pull request by fti-rchaniya**
**Description**
This PR has following change in the Area Upload Manager component:

- While using this component, if an user wants to set selected files to null by an external triggered event such as close icon or button as shown in below attached screenshot, then the selected files does not reset to null due to the absence of prop which setFiles(null).
image

- Due to this, even if user click on close icon, and add files again so the expected behaviour is the selected file's count start from 0 which is not, it starts adding in the array where the previously added files are saved. 
- For more understanding, kindly refer below screencast:
https://www.awesomescreenshot.com/video/16183735?key=219730f4f27ff7f0ec403ed16e3e3611

**Considerations on Implementation**

- To achieve this, earlier we thought to add 1 Boolean state to the component which an user can use as a prop to setFiles(null) as shown in below attached screenshot.
image

- Instead of the above solution, we approaches another suggestion given in this [comment](https://github.com/future-standard/scorer-ui-kit/pull/407#pullrequestreview-1377390065).
- As per the above comment, instead of adding prop to trigger external event, we have added a "Clear Files" button in the component itself to clear selected files.
- Also "Clear Files" button will be only visible if the selected files count is more than 0, it is aligned parallelly with the "Select Files"/"Add More Files" button as shown in the below attached screenshot:
image

Screencast of the final implementation:
https://www.awesomescreenshot.com/video/16402979?key=eaec76b4966083b3e92b269a375fde97

### CODE REVIEW

File: src/Form/organisms/AreaUploadManager.tsx [hunk 1]
```diff
@@ -105,6 +105,7 @@ interface IAreaUploaderManager {
   allowedFileTypes?: string[]
   customComponent?: ReactElement
   onChangeCallback?: (goodFiles: FileList, rejectedFiles: FileList) => void
+  clearState?: boolean;
```
Comment (fti-smaske) [inline src/Form/organisms/AreaUploadManager.tsx:108]: Change varaible name to `clearFiles`

**Review by fti-skumar — APPROVED** [review]
LGTM, just change the name as suggested by @fti-smaske

File: src/Form/organisms/AreaUploadManager.tsx (hunk 1 above)
Comment (fti-rchaniya) [reply src/Form/organisms/AreaUploadManager.tsx]: Done, thanks for the feedback

**Review by fti-smaske — APPROVED** [review]

File: storybook/src/stories/Form/FileManagement/AreaUploadManager.stories.tsx [hunk 2]
```diff
@@ -27,6 +27,7 @@ export const _AreaUploadManager = () => {
   const fileIcons = object('File Icons', defaultIcons)
   const showValue = action('Input Callback');
   const allowedFileTypes = object('allowedFileTypes', fileTypes);
+  let clearState = boolean('Clear selected Files', false);
```
Comment (fti-bcarpenter) [inline storybook/src/stories/Form/FileManagement/AreaUploadManager.stories.tsx:30]: This should be `clearFiles`. Isn't it?

File: storybook/src/stories/Form/FileManagement/AreaUploadManager.stories.tsx [hunk 3]
```diff
@@ -39,6 +40,7 @@ export const _AreaUploadManager = () => {
       <AreaUploadManager
         {...{ selectFilesText, title, fileIcons, description, allowedFileTypes, addMoreFilesText }}
         onChangeCallback={inputCallback}
+        clearState={clearState}
```
Comment (fti-bcarpenter) [inline storybook/src/stories/Form/FileManagement/AreaUploadManager.stories.tsx:43]: This should be `clearFiles`. Isn't it?

**Review by fti-bcarpenter — CHANGES_REQUESTED** [review]

**Review by fti-bcarpenter — APPROVED** [review]
LGTM

**Review by atomworks — CHANGES_REQUESTED** [review]
Could we move the button to clear this in to the AreaUploadManager component itself? I think we can just show it all the time as it's useful. 

Just one of the grey buttons, small size and centered in the container would be great.

**Comment by fti-rchaniya** [comment]
> Could we move the button to clear this in to the AreaUploadManager component itself? I think we can just show it all the time as it's useful.
> 
> Just one of the grey buttons, small size and centered in the container would be great.

Yes, we can. Can you please confirm me the place where we can place that button to clear files

**Comment by fti-rchaniya** [comment]
> Could we move the button to clear this in to the AreaUploadManager component itself? I think we can just show it all the time as it's useful.
> 
> Just one of the grey buttons, small size and centered in the container would be great.

Thanks for the feedback, suggested changes have been implemented

**Review by atomworks — APPROVED** [review]

**Pull request merged**